### PR TITLE
JAMES-2153 Added docker file for Guice JPA-SMTP

### PR DESCRIPTION
**Pull request by matzepan**
Added a docker file with the default settings as described in [Guice JPA SMTP Guide](https://james.apache.org/searver/install/guice-jpa-smtp.html) for [JAMES-2153](https://issues.apache.org/jira/browse/JAMES-2153)

### CODE REVIEW

**Comment by chibenwa** [comment]
I am relaunching the CI build on your pull request

test this please

File: dockerfiles/run/guice/jpa-smtp/destination/conf/jwt_publickey [hunk 1]
```diff
@@ -0,0 +1,9 @@
+-----BEGIN PUBLIC KEY-----
```
[the remark this answers is not in the document]
Comment (matzepan) [reply dockerfiles/run/guice/jpa-smtp/destination/conf/jwt_publickey]: @chibenwa I was using the examples from https://github.com/apache/james-project/tree/master/server/container/guice/jpa-smtp/sample-configuration - I'm not that familiar to actually know what settings actually are needed.

If you tell me what are the files that don't make sense and if the sample configuration is stale regarding the current version I can update this pull request.

File: dockerfiles/run/guice/jpa-smtp/destination/conf/mailetcontainer.xml [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+
+        <processor state="transport" enableJmx="true">
+            <matcher name="relay-allowed" match="org.apache.james.mailetcontainer.impl.matchers.Or">
+                <matcher match="SMTPAuthSuccessful"/>
+                <matcher match="SMTPIsAuthNetwork"/>
+                <matcher match="SentByMailet"/>
+            </matcher>
+
+            <mailet match="All" class="RecipientRewriteTable" />
+            <mailet match="RecipientIsLocal" class="LocalDelivery"/>
```
[the remark this answers is not in the document]
Comment (matzepan) [reply dockerfiles/run/guice/jpa-smtp/destination/conf/mailetcontainer.xml]: No this container doesn't start - as you said there is an error that there is no MailboxManager present. I used https://github.com/apache/james-project/blob/master/server/container/guice/jpa-smtp/sample-configuration/mailetcontainer.xml here.

File: dockerfiles/run/guice/jpa-smtp/destination/conf/jwt_publickey (hunk 1 above)
[the remark this answers is not in the document]
Comment (chibenwa) [reply dockerfiles/run/guice/jpa-smtp/destination/conf/jwt_publickey]: This file can be removed.

File: dockerfiles/run/guice/jpa-smtp/destination/conf/mailetcontainer.xml (hunk 2 above)
[the remark this answers is not in the document]
Comment (chibenwa) [reply dockerfiles/run/guice/jpa-smtp/destination/conf/mailetcontainer.xml]: Please do remove Local delivery then.

About sample-configuration, that folder should be removed and the documentation (readme in /server/container/guice/jpa-smtp*) should rely on the conf files you introduced for this docker IMO

[the remark this answers is not in the document]
Comment (matzepan) [reply dockerfiles/run/guice/jpa-smtp/destination/conf/mailetcontainer.xml]: @chibenwa I removed the Local delivery line, and fixed another Problem with BCCs.

File: dockerfiles/run/guice/jpa-smtp/destination/conf/jwt_publickey (hunk 1 above)
[the remark this answers is not in the document]
Comment (matzepan) [reply dockerfiles/run/guice/jpa-smtp/destination/conf/jwt_publickey]: I removed it.

File: dockerfiles/run/guice/jpa-smtp/destination/conf/usersrepository.xml [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+  software distributed under the License is distributed on an  
+  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       
+  KIND, either express or implied.  See the License for the    
+  specific language governing permissions and limitations      
+  under the License.                                           
+ -->
+
+<usersrepository name="LocalUsers" class="org.apache.james.user.jpa.JPAUsersRepository">
+    <destination URL="file://users/"/>
+    <algorithm>MD5</algorithm>
```
[the remark this answers is not in the document]
Comment (matzepan) [reply dockerfiles/run/guice/jpa-smtp/destination/conf/usersrepository.xml]: Changed it to SHA-512

File: dockerfiles/compilation/java-8/compile.sh [hunk 4]
```diff
@@ -15,6 +15,7 @@ printUsage() {
 ORIGIN=/origin
 CASSANDRA_DESTINATION=/cassandra/destination
 JPA_DESTINATION=/jpa/destination
+JPA_SMPT_DESTINATION=/jpa-smpt/destination
```
Comment (rouazana) [inline dockerfiles/compilation/java-8/compile.sh]: s/SMPT/SMTP/

File: dockerfiles/compilation/java-8/compile.sh [hunk 5]
```diff
@@ -77,6 +78,14 @@ if [ $? -eq 0 ]; then
       cp -r server/container/cli/target/james-server-cli.lib $JPA_DESTINATION || true
    fi
 
+   if [ -d "$JPA_SMPT_DESTINATION" ]; then
```
Comment (rouazana) [inline dockerfiles/compilation/java-8/compile.sh]: idem

File: dockerfiles/compilation/java-8/compile.sh [hunk 6]
```diff
@@ -77,6 +78,14 @@ if [ $? -eq 0 ]; then
...
+      echo "Copying JPA-SMPT jars"
```
Comment (rouazana) [inline dockerfiles/compilation/java-8/compile.sh]: idem

File: dockerfiles/run/guice/jpa-smtp/destination/extensions-jars/README.md [hunk 7]
```diff
@@ -0,0 +1,5 @@
+# Adding Jars to JAMES
+
+The jar in this folder will be added to JAMES classpath when mounted under /root/extensions-jars inside the running container.
+
+You can use it to add you customs Mailets/Matchers.
```
Comment (rouazana) [inline dockerfiles/run/guice/jpa-smtp/destination/extensions-jars/README.md]: s/you customs/your custom/

**Comment by chibenwa** [comment]
Hi,

This pull request has just been merged.

Thank you again for your dedication toward the James project. If you need help, want to [contribute](https://issues.apache.org/jira/issues/?jql=project%20%3D%20JAMES%20AND%20resolution%20%3D%20Unresolved%20AND%20labels%20%3D%20newbie%20ORDER%20BY%20priority%20DESC%2C%20updated%20DESC) [documentation](https://issues.apache.org/jira/issues/?jql=project%20%3D%20JAMES%20AND%20resolution%20%3D%20Unresolved%20AND%20labels%20%3D%20documentation%20ORDER%20BY%20priority%20DESC%2C%20updated%20DESC), new [features](https://issues.apache.org/jira/issues/?jql=project%20%3D%20JAMES%20AND%20resolution%20%3D%20Unresolved%20AND%20labels%20%3D%20feature%20ORDER%20BY%20priority%20DESC%2C%20updated%20DESC), tests or [bugfix](https://issues.apache.org/jira/issues/?jql=project%20%3D%20JAMES%20AND%20resolution%20%3D%20Unresolved%20AND%20labels%20%3D%20easyfix%20ORDER%20BY%20priority%20DESC%2C%20updated%20DESC), please don't hesitate to reach us.

Thanks again,

Cheers,

Benoit

**Pull request closed without merging**